### ISSUE TITLE
`dolt_commit()` will break some following statements in stored procedures

### ISSUE THREAD

**Issue by zachmu** [dolthub/dolt#5829]
`call dolt_commit()` commits the current SQL transaction and doesn't notify the SQL engine about it in a well-controlled way. This is fine in the context of normal statements: the engine notices there's no active transaction and starts one. But in a stored procedure, anystatement following `call dolt_commit()` will behave strangely / fail because there isn't an active transaction.

**Comment by zachmu** [comment]
Pending work extends this bug to `dolt_branch()` as well, and more stored procedures will follow as we make the cross-branch transaction semantics more coherent.

**Comment by timsehn** [comment]
I just talked to @zachmu about this. He seems to remember it being tricky but he's going to dig in and scope it now.

**Comment by zachmu** [comment]
I think a hacky version of this might be pretty near at hand. I'll have a prototype later this week.

**Comment by zachmu** [comment]
I have this working for most things, but tests have revealed a deeper issue: because of how we analyze and execute stored procedures, there are certain sequences that won't work properly without a large rewrite. This includes things like `dolt_checkout()`, since the effective branch of the insert was resolved before execution of the procedure began. Schema updates during procedure execution will be problematic for similar reasons. I'll update this issue when I've scoped the problem more precisely.
